### DISCUSSION TITLE
View how many times a morph appears in total across all cards

### DISCUSSION THREAD

**Discussion by ghost** [mortii/anki-morphs#231] [Ideas]
When importing cards from subs2srs or similar, there will be a lot of cards, and some of them will have morphs that appear very few times.  For example, i have found one card in my deck with the morph '牛角尖'.  I think it would be useful to be able to see all the morphs that ankimorphs has found with their frequency across cards.  I would then be able to suspend or delete these cards that are probably unproductive to study and mostly serve to make syncing take longer and put more strain on the ankiweb server.

**Comment by mortii** [comment]
That sounds interesting. How and/or where would you like it to be displayed?

**Comment by ghost** [comment]
I'm not entirely sure, because i think that perhaps it would work best as a new keyword in the browse screen which seems like a lot of work for what is likely a niche feature.  I think i would get the same benefit from some kind of option like `Tag cards with morphs which appear X or fewer times`.  

I'm interested in hearing if anyone else also would get any use from that?

**Comment by mortii** [comment]
One potential solution would be to make the ["known morphs exporter"](https://mortii.github.io/anki-morphs/user_guide/usage/known-morphs-exporter.html) output files be more similar to the frequency files by adding the "occurrence" row. 

![image](https://github.com/mortii/anki-morphs/assets/15674619/bd16f393-1e3d-4c4a-b926-b27b375ef69a)

You could then set the known morph interval to zero to see all the morphs that are in the database and their occurrences.

**Comment by ghost** [comment]
I think that this solution would work quite well for what i want.  It also reminded me that maybe i can achieve the same thing by doing a deck plain text export, removing the unwanted columns, and passing it to the frequency file generator.  If this works for me, i suppose with no other interest in the feature, there's no need to implement that.

That said, it'd be nice to have the option to have more flexibility with the known morphs files.  Sometimes i will export morphs from other programs (usually language crush which i use for reading), and i have to do manipulation before i can put them in the known folder.  This isn't too difficult, but making it more streamlined might be nice.

**Reply by mortii** [reply]
>That said, it'd be nice to have the option to have more flexibility with the known morphs files. Sometimes i will export morphs from other programs (usually language crush which i use for reading), and i have to do manipulation before i can put them in the known folder. This isn't too difficult, but making it more streamlined might be nice.

What specific kind of flexibility and/or streamlining do you want?

**Reply by ghost** [reply]
I successfully got the output i want using the method i outlined above.  Having done that, i think that perhaps this feature is not worth implementing, because (at least for chinese) the morphemiser is not of a high enough quality for me not to unintentionally remove a lot of useful cards.

The morphemiser appears to favour longer strings of characters than shorter, resulting in things like 前 or 岁 often being appended to already known morphs and registered as a new morph.  In practice, this doesn't cause an issue as there are enough other cards available that i don't feel that i'm missing out on sentences, but i also don't want to just delete them because some of them have lesser known but useful to study words on them.

**Reply by mortii** [reply]
> I successfully got the output i want using the method i outlined above. Having done that, i think that perhaps this feature is not worth implementing, because (at least for chinese) the morphemiser is not of a high enough quality for me not to unintentionally remove a lot of useful cards.

I think an occurrence column could be interesting if nothing else, so I do want to add it, but it won't be super high priority and might therefore take a while.

> The morphemiser appears to favour longer strings of characters than shorter, resulting in things like 前 or 岁 often being appended to already known morphs and registered as a new morph. In practice, this doesn't cause an issue as there are enough other cards available that i don't feel that i'm missing out on sentences, but i also don't want to just delete them because some of them have lesser known but useful to study words on them.

Have you tried any of the spaCy morphemizers? They split words more aggressively which some people prefer. That being said, all morphemizers have their own idiosyncrasies and problems, so it boils down to preference and choosing the lesser of two evils.

**Reply by ghost** [reply]
I'm in the process of trying to get spaCy installed now.  I'm on debian, and the deadsnakes repo doesn't work for me, so i'm currently building python3.9 from source.  Will see how it goes!

**Reply by mortii** [reply]
I installed ubuntu 24.04 yesterday and got deadsnakes working, but yes, it doesn't work for every distro version. You could also try pyvenv, it's much simpler than building from source imo.

**Comment by mortii** [comment]
The known morphs exporter now includes an optional occurrences column: https://github.com/mortii/anki-morphs/releases/tag/v3.0.0
